addons/ness-lambda/@{{ cookiecutter.root }}/@{{ cookiecutter.lambda_name }}.tf

```diff
@@ -0,0 +1,181 @@
+locals {
+  # Additional policies to attach to the lambda
+  lambda_@{{ cookiecutter.lambda_underscore }}_policies = [
+    {% if cookiecutter.is_triggered_by_sqs == "true" -%}
+    "arn:aws:iam::aws:policy/service-role/AWSLambdaSQSQueueExecutionRole", # Needed to allow Lambda manage sqs, see https://docs.aws.amazon.com/lambda/latest/dg/services-sqs-configure.html#events-sqs-permissions
+    {% endif -%}
+  ]
+}
+
+variable "@{{ cookiecutter.lambda_underscore }}" {
+  description = "Configuration of the @{{ cookiecutter.lambda_name }} workflow"
+  type = object({
+    {% if cookiecutter.is_triggered_by_sqs == "true" -%}
+    sqs_batch_size                        = number
+    {% endif -%}
+    cloudwatch_logs_retention_in_days     = number
+    lambda_reserved_concurrent_executions = number
+    lambda_timeout                        = number
+  })
+  default = {
+    {% if cookiecutter.is_triggered_by_sqs == "true" -%}
+    sqs_batch_size                        = 10
+    {% endif -%}
+    cloudwatch_logs_retention_in_days     = 7
+    lambda_reserved_concurrent_executions = 2
+    lambda_timeout                        = 60
+  }
+}
+
+module "@{{ cookiecutter.lambda_underscore }}" {
+  source  = "terraform-aws-modules/lambda/aws"
+  version = "~> 7.0"
+
+  function_name          = "${var.environment}-@{{ cookiecutter.lambda_name }}"
+  description            = "@{{ cookiecutter.description }}"
+  source_path            = "./lambdas/@{{ cookiecutter.lambda_name }}"
```

Comment:
> I would use `lambda_underscore` since we expect the name to potentially contain spaces.

name should not contain spaces. If they put, the workflow will fail and they will need to fix